ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -243,6 +244,7 @@ type EdgeStatusVar m = StrictTVar m EdgeStatus
 -- each node.
 runThreadNetwork :: forall m blk.
                     ( IOLike m
+                    , MonadTimer m
```

Comment:
Can we make `MonadTimer` part of `IOLike`?

Reply:
I see that as consensus question, `IOLike` is not used by networking.

Reply:
Yeah sure, but you're touching consensus code here. Consider it as me asking to take off your shoes in my house :slightly_smiling_face:. 

If you want, I can push a commit making the change to this PR tomorrow.

Reply:
> If you want, I can push a commit making the change to this PR tomorrow.

Please do.
The number of TODOs added in `NodeNetwork.hs` is an indication that I really didn't want to barge into your house and connect a typed-protocol gas detector to your electric stove. :)

Reply:
I (force-)pushed my changes to this branch.

You'll need to explain those TODOs that you left in `NodeNetwork.hs`.

Reply:
Thanks.
This PR will implement #1395 which includes per protocol timeouts but not byte limits.
Byte limits could have been trivial to implement, (`(byteLimitsBlockFetch (fromIntegral . LBS.length))`), if it wasn't for the 8 different byte types used by `consensusNetworkApps`. They will be implemented in a future PR.

Reply:
After discussion with @mrBliss MonadTimer will not be added to IOLike so his commit will be removed.